core/core-funcs.el

```diff
@@ -36,12 +36,18 @@
   (run-hooks 'text-mode-hook))
 
 (defun spacemacs//get-package-directory (pkg)
-  "Return the directory of PKG. Return nil if not found."
-  (let ((elpa-dir (file-name-as-directory package-user-dir)))
-    (when (file-exists-p elpa-dir)
-      (let* ((pkg-match (concat (symbol-name pkg) "-[0-9]+"))
-             (dir (car (directory-files elpa-dir 'full pkg-match))))
-        (when dir (file-name-as-directory dir))))))
+  "Return the installed directory of PKG. Return `nil' if not found."
+  (let ((directories (cons package-user-dir
```

Comment:
If spacemacs used package-alist to figure out package installation directory instead of searching package-user-dir,  then I think this PR would not be needed.  All I would have to do is just install packages under site-lisp, and package.el would take care of things.